samples/cpp/embedded-speech/samples/settings.cpp

```diff
@@ -238,7 +238,7 @@ bool VerifySettings()
                     }
                     else
                     {
-                        return model->Name.compare(modelName) == 0;
+                        return model->Name.compare(modelName) == 0 || model->Locales[0].compare(modelName) == 0;
```

Comment:
>[0] [](http://example.com/codeflow?start=84&length=3)

Seems risky to assume a hard coded index.
Also, what happens if locales is empty?

Reply:
Model information comes from `GetSpeechRecognitionModels()`. SDK requires that a model has at least one locale (non-empty string), otherwise it is ignored when scanning for models. So model->Locales[0] is always there.